code/modules/martial_arts/cqc.dm

```diff
@@ -0,0 +1,186 @@
+#define SLAM_COMBO "HG"
+#define KICK_COMBO "HH"
+#define RESTRAIN_COMBO "GG"
+#define PRESSURE_COMBO "DG"
+#define CONSECUTIVE_COMBO "DDH"
+
+/datum/martial_art/cqc
+	name = "CQC"
+	help_verb = /mob/living/carbon/human/proc/CQC_help
+	block_chance = 75
+
+/datum/martial_art/cqc/proc/drop_restraining()
+	restraining = FALSE
+
+/datum/martial_art/cqc/proc/check_streak(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	if(findtext(streak,SLAM_COMBO))
+		streak = ""
+		Slam(A, D)
+		return TRUE
+	if(findtext(streak,KICK_COMBO))
+		streak = ""
+		Kick(A, D)
+		return TRUE
+	if(findtext(streak,RESTRAIN_COMBO))
+		streak = ""
+		Restrain(A, D)
+		return TRUE
+	if(findtext(streak,PRESSURE_COMBO))
+		streak = ""
+		Pressure(A, D)
+		return TRUE
+	if(findtext(streak,CONSECUTIVE_COMBO))
+		streak = ""
+		Consecutive(A, D)
+	return FALSE
+
+/datum/martial_art/cqc/proc/Slam(mob/living/carbon/human/A, mob/living/carbon/human/D)
```

Comment:
on TG it does have logging, consider adding attack logs on the manuvers.

Reply:
Please do